client/shared/src/api/client/types/hover.ts

```diff
@@ -26,6 +30,15 @@ export function fromHoverMerged(values: (Badged<Hover | PlainHover> | null | und
             if (result.alerts) {
                 alerts.push(...result.alerts)
             }
+
+            const tags = [result as Badged<object>]
+                .concat(result.alerts || [])
+                .flatMap(badged => badged.aggregableTags || [])
```

Comment:
I believe if we do `[result, ...(result.alerts || [])]` instead of `[result].concat(result.alerts || [])` we need a type assertion 
```suggestion
            const tags = [result, ...(result.alerts || [])].flatMap(badged => badged.aggregableTags || [])
```